lease/lessor.go

```diff
@@ -124,6 +124,8 @@ type lessor struct {
 	// a raft follow is the primary between the raft leader demotion and lessor demotion.
 	// Usually this should not be a problem. Lease should not be that sensitive to timing.
 	primary bool
+	// demotec will be closed once the lessor get demoted
+	demotec chan struct{}
```

Comment:
but it means it's tracking the same state twice. How about `isPrimary() bool { return demotec != nil }`?